src/service/video-manager-impl.js

```diff
@@ -284,20 +303,51 @@ export class VideoManager {
     return this.getEntryForVideo_(video).userInteractedWithAutoPlay();
   }
 
+  /**
+   * Checks whether there's no video already docked
+   *
+   * @param {VideoEntry} entry
+   * @return {boolean}
+   */
+  canDock(entry) {
+    return !this.dockedVideo_ || this.dockedVideo_ == entry;
+  }
+
+  /**
+   * Registers the provided video as docked
+   *
+   * @param {VideoEntry} entry
+   */
+  registerDocked(entry) {
+    this.dockedVideo_ = entry;
+  }
+
+  /**
+   * Un-registers the currently docked video
+   */
+  unregisterDocked() {
+    this.dockedVideo_ = null;
+    for (let i = 0; i < this.entries_.length; i++) {
+      this.entries_[i].hasBeenInViewBefore = false;
```

Comment:
This logic fixes a very particular scenario where you have two or more videos on the same page, you dock one, undock it then scroll back to the other video which begins to undock although it has never been docked (see GIF). We needed a restriction to only dock videos when they have been fully in view before. If you can think of a better logic that doesn't require a variable, please let me know.
![ezgif com-video-to-gif 8](https://user-images.githubusercontent.com/591655/28145769-b669c50a-6729-11e7-8533-39903f3708df.gif)